buster/tokenizers/base.py

```diff
@@ -12,10 +12,10 @@ def encode(self, string: str) -> list[int]:
         ...
 
     @abstractmethod
-    def decode(self, encoded: list[int]):
+    def decode(self, encoded: list[int]) -> str:
         ...
 
-    def num_tokens(self, string: str, return_encoded: bool = False):
+    def num_tokens(self, string: str, return_encoded: bool = False) -> int:
```

Comment:
should be `Union[int, tuple[int, str]]` no?

Reply:
ah youre right